UI/ObjectListWnd.cpp

```diff
@@ -1392,7 +1392,7 @@ class ObjectPanel : public GG::Control {
 
         for (std::size_t i = 1; i < m_controls.size(); ++i) {
             right = left + GetColumnWidth(static_cast<int>(i));
-            if (ctrl = m_controls[i])
+            if ((ctrl = m_controls[i]))
```

Comment:
The changes in this PR should not generate different code.  They only make the spurious compiler warnings go away.

The compiler wants the two operations, assignment to `ctrl` and testing of the boolean clause in separately bracketed clauses.  Hence the extra set of brackets.